--- a/BUILD.bazel	2019-08-30 15:45:54.644313954 -0700
+++ b/BUILD.bazel	2019-08-30 15:46:17.214156324 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -13,3 +13,10 @@
     embed = [":go_default_library"],
     deps = ["@com_github_go_toolsmith_strparse//:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["astfmt.go"],
+    importpath = "github.com/go-toolsmith/astfmt",
+    visibility = ["//visibility:public"],
+)
