--- a/BUILD.bazel	2019-09-04 16:22:31.596304211 -0700
+++ b/BUILD.bazel	2019-09-04 16:22:57.532699847 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -31,3 +31,21 @@
         "@com_github_davecgh_go_spew//spew:go_default_library",
     ],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "decoder.go",
+        "hcl.go",
+        "lex.go",
+        "parse.go",
+    ],
+    importpath = "github.com/hashicorp/hcl",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//hcl/ast:go_tool_library",
+        "//hcl/parser:go_tool_library",
+        "//hcl/token:go_tool_library",
+        "//json/parser:go_tool_library",
+    ],
+)
--- a/hcl/printer/BUILD.bazel	2019-09-04 16:28:12.316595218 -0700
+++ b/hcl/printer/BUILD.bazel	2019-09-04 16:28:33.950361652 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -22,3 +22,18 @@
     embed = [":go_default_library"],
     deps = ["//hcl/parser:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "nodes.go",
+        "printer.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/hcl/printer",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//hcl/ast:go_tool_library",
+        "//hcl/parser:go_tool_library",
+        "//hcl/token:go_tool_library",
+    ],
+)
--- a/hcl/ast/BUILD.bazel	2019-09-05 13:13:09.951487587 -0700
+++ b/hcl/ast/BUILD.bazel	2019-09-05 13:13:30.985205026 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -17,3 +17,14 @@
     embed = [":go_default_library"],
     deps = ["//hcl/token:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "ast.go",
+        "walk.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/hcl/ast",
+    visibility = ["//visibility:public"],
+    deps = ["//hcl/token:go_tool_library"],
+)
--- a/hcl/token/BUILD.bazel	2019-09-05 13:35:05.567286302 -0700
+++ b/hcl/token/BUILD.bazel	2019-09-05 13:35:27.201052735 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -16,3 +16,14 @@
     srcs = ["token_test.go"],
     embed = [":go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "position.go",
+        "token.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/hcl/token",
+    visibility = ["//visibility:public"],
+    deps = ["//hcl/strconv:go_tool_library"],
+)
--- a/hcl/strconv/BUILD.bazel	2019-09-05 13:37:23.175933040 -0700
+++ b/hcl/strconv/BUILD.bazel	2019-09-05 13:37:41.865459071 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -12,3 +12,10 @@
     srcs = ["quote_test.go"],
     embed = [":go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["quote.go"],
+    importpath = "github.com/hashicorp/hcl/hcl/strconv",
+    visibility = ["//visibility:public"],
+)
--- a/hcl/parser/BUILD.bazel	2019-09-05 13:38:36.889951914 -0700
+++ b/hcl/parser/BUILD.bazel	2019-09-05 13:39:04.536209278 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -27,3 +27,18 @@
         "//hcl/token:go_default_library",
     ],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "error.go",
+        "parser.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/hcl/parser",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//hcl/ast:go_tool_library",
+        "//hcl/scanner:go_tool_library",
+        "//hcl/token:go_tool_library",
+    ],
+)
--- a/hcl/scanner/BUILD.bazel	2019-09-05 13:39:49.363869533 -0700
+++ b/hcl/scanner/BUILD.bazel	2019-09-05 13:40:13.005799936 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -14,3 +14,11 @@
     embed = [":go_default_library"],
     deps = ["//hcl/token:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["scanner.go"],
+    importpath = "github.com/hashicorp/hcl/hcl/scanner",
+    visibility = ["//visibility:public"],
+    deps = ["//hcl/token:go_tool_library"],
+)
--- a/json/parser/BUILD.bazel	2019-09-05 13:41:01.078363550 -0700
+++ b/json/parser/BUILD.bazel	2019-09-05 13:41:21.497815972 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -25,3 +25,19 @@
         "//hcl/token:go_default_library",
     ],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "flatten.go",
+        "parser.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/json/parser",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//hcl/ast:go_tool_library",
+        "//hcl/token:go_tool_library",
+        "//json/scanner:go_tool_library",
+        "//json/token:go_tool_library",
+    ],
+)
--- a/json/scanner/BUILD.bazel	2019-09-05 13:42:04.675108514 -0700
+++ b/json/scanner/BUILD.bazel	2019-09-05 13:42:26.080856330 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -14,3 +14,11 @@
     embed = [":go_default_library"],
     deps = ["//json/token:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["scanner.go"],
+    importpath = "github.com/hashicorp/hcl/json/scanner",
+    visibility = ["//visibility:public"],
+    deps = ["//json/token:go_tool_library"],
+)
--- a/json/token/BUILD.bazel	2019-09-05 13:43:01.807773493 -0700
+++ b/json/token/BUILD.bazel	2019-09-05 13:43:22.677477540 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -16,3 +16,14 @@
     srcs = ["token_test.go"],
     embed = [":go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "position.go",
+        "token.go",
+    ],
+    importpath = "github.com/hashicorp/hcl/json/token",
+    visibility = ["//visibility:public"],
+    deps = ["//hcl/token:go_tool_library"],
+)
