--- a/BUILD.bazel	2019-09-04 16:29:30.650991355 -0700
+++ b/BUILD.bazel	2019-09-04 16:29:49.852559195 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -22,3 +22,14 @@
     ],
     embed = [":go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "decode_hooks.go",
+        "error.go",
+        "mapstructure.go",
+    ],
+    importpath = "github.com/mitchellh/mapstructure",
+    visibility = ["//visibility:public"],
+)
