--- a/BUILD.bazel	2019-08-26 16:18:34.933867435 -0700
+++ b/BUILD.bazel	2019-08-26 16:19:32.309865515 -0700
@@ -1,10 +1,20 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_tool_library")
 
 go_library(
     name = "go_default_library",
     srcs = [
         "parser.go",
         "visitor.go",
+    ],
+    importpath = "github.com/golangci/goconst",
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "parser.go",
+        "visitor.go",
     ],
     importpath = "github.com/golangci/goconst",
     visibility = ["//visibility:public"],
