--- a/BUILD.bazel	2019-09-03 11:23:13.964127999 -0700
+++ b/BUILD.bazel	2019-09-03 11:23:50.963149032 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -20,3 +20,17 @@
     embed = [":go_default_library"],
     deps = ["@com_github_mattn_go_colorable//:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "color.go",
+        "doc.go",
+    ],
+    importpath = "github.com/fatih/color",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_github_mattn_go_colorable//:go_tool_library",
+        "@com_github_mattn_go_isatty//:go_tool_library",
+    ],
+)
