--- a/BUILD.bazel	2019-09-04 16:14:52.970356223 -0700
+++ b/BUILD.bazel	2019-09-04 16:15:13.884063863 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -12,3 +12,10 @@
     srcs = ["homedir_test.go"],
     embed = [":go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["homedir.go"],
+    importpath = "github.com/mitchellh/go-homedir",
+    visibility = ["//visibility:public"],
+)
