--- a/go/loader/BUILD.bazel	
+++ b/go/loader/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test","go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -16,6 +16,23 @@ go_library(
     ],
 )
 
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "doc.go",
+        "loader.go",
+        "util.go",
+    ],
+    importpath = "golang.org/x/tools/go/loader",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//go/ast/astutil:go_tool_library",
+        "//go/buildutil:go_tool_library",
+        "//go/internal/cgo:go_tool_library",
+    ],
+)
+
+
 go_test(
     name = "go_default_test",
     srcs = [
