--- a/width/BUILD.bazel	2019-08-29 15:53:50.570938280 -0700
+++ b/width/BUILD.bazel	2019-08-29 15:54:23.009583084 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -32,3 +32,18 @@
         "//transform:go_default_library",
     ],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "kind_string.go",
+        "tables10.0.0.go",
+        "tables9.0.0.go",
+        "transform.go",
+        "trieval.go",
+        "width.go",
+    ],
+    importpath = "golang.org/x/text/width",
+    visibility = ["//visibility:public"],
+    deps = ["//transform:go_tool_library"],
+)
--- a/transform/BUILD.bazel	2019-08-29 15:56:30.501021906 -0700
+++ b/transform/BUILD.bazel	2019-08-29 15:57:01.255529398 -0700
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -19,3 +19,10 @@
         "//unicode/norm:go_default_library",
     ],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = ["transform.go"],
+    importpath = "golang.org/x/text/transform",
+    visibility = ["//visibility:public"],
+)
